api/admin/controller/report.py

```diff
@@ -0,0 +1,42 @@
+import json
```

Comment:
Minor: I've been trying to slowly move our code towards being able to run with mypy in strict mode. I think this code will pass those tests with almost no changes, so it would be nice to add this new file to `pyproject.toml` in the `[[tool.mypy.overrides]]` section that enables strict mode.

Reply:
Re the SQL query:  there is a test covering it, so if there is a problem with the model, the test will fail.